x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/preview_rules_route.ts

```diff
@@ -224,6 +226,14 @@ export const previewRulesRoute = async (
             startedAt.add(parseInterval(internalRule.schedule.interval));
             invocationCount--;
           }
+
+          if (isAborted) {
+            logs.push({
+              warnings: [`Preview timed out after ${PREVIEW_TIMEOUT_SECONDS} seconds`],
+              errors: [],
+              startedAt: moment().toDate().toISOString(),
```

Comment:
The `logs` structure contains an array of objects where each object corresponds to a particular rule invocation, so adding an object that is not from a rule invocation mixes different concepts together. Can we return this warning as a separate object in the response body instead of including it in `logs`?